packages/flutter/test/gestures/positioned_details_test.dart

```diff
@@ -0,0 +1,91 @@
+        globalPosition: globalPosition,
+        localPosition: localPosition,
+        pressure: 0,
+      ),
+      (Offset globalPosition, {Offset? localPosition}) =>
+          LongPressDownDetails(globalPosition: globalPosition, localPosition: localPosition),
+      (Offset globalPosition, {Offset? localPosition}) =>
+          LongPressStartDetails(globalPosition: globalPosition, localPosition: localPosition),
+      (Offset globalPosition, {Offset? localPosition}) =>
+          LongPressMoveUpdateDetails(globalPosition: globalPosition, localPosition: localPosition),
+      (Offset globalPosition, {Offset? localPosition}) =>
+          LongPressEndDetails(globalPosition: globalPosition, localPosition: localPosition),
+      (Offset globalPosition, {Offset? localPosition}) => SerialTapDownDetails(
+        globalPosition: globalPosition,
+        localPosition: localPosition,
+        kind: PointerDeviceKind.unknown,
+      ),
+      (Offset globalPosition, {Offset? localPosition}) =>
+          SerialTapUpDetails(globalPosition: globalPosition, localPosition: localPosition),
+      (Offset globalPosition, {Offset? localPosition}) =>
+          TapDownDetails(globalPosition: globalPosition, localPosition: localPosition),
+      (Offset globalPosition, {Offset? localPosition}) => TapUpDetails(
+        globalPosition: globalPosition,
+        localPosition: localPosition,
+        kind: PointerDeviceKind.unknown,
+      ),
+      (Offset globalPosition, {Offset? localPosition}) => TapDragDownDetails(
+        globalPosition: globalPosition,
+        localPosition: localPosition ?? globalPosition,
+        consecutiveTapCount: 0,
+      ),
+      (Offset globalPosition, {Offset? localPosition}) => TapDragUpDetails(
+        globalPosition: globalPosition,
+        localPosition: localPosition ?? globalPosition,
+        kind: PointerDeviceKind.unknown,
+        consecutiveTapCount: 0,
+      ),
+      (Offset globalPosition, {Offset? localPosition}) => TapDragStartDetails(
+        globalPosition: globalPosition,
+        localPosition: localPosition ?? globalPosition,
+        consecutiveTapCount: 0,
+      ),
+      (Offset globalPosition, {Offset? localPosition}) => TapDragUpdateDetails(
+        globalPosition: globalPosition,
+        localPosition: localPosition ?? globalPosition,
+        consecutiveTapCount: 0,
+        offsetFromOrigin: Offset.zero,
+        localOffsetFromOrigin: Offset.zero,
+      ),
+      (Offset globalPosition, {Offset? localPosition}) => TapDragEndDetails(
+        globalPosition: globalPosition,
+        localPosition: localPosition ?? globalPosition,
+        consecutiveTapCount: 0,
+      ),
+    };
+
+    for (final _Create create in creates) {
+      final PositionedGestureDetails details1 = create(const Offset(0, 100));
+      expect(details1.globalPosition, const Offset(0, 100));
+      expect(details1.localPosition, const Offset(0, 100));
+      final PositionedGestureDetails details2 = create(
```

Comment:
The second half of the test doesn't really help much; it merely verifies that the `create` functions created by this test file default local positions to global positions. 

I'm not sure how this PR can be tested. Maybe we can omit tests, especially if we adopt the `abstract interface` design.